### ISSUE TITLE
[Bug]: Problem in datepicker keyboard navigation.

### ISSUE THREAD

**Issue by ankurrsinghal** [appsmithorg/appsmith#12406]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Property Pane `Default Date` control uses Blueprintjs [DateInput](https://blueprintjs.com/docs/#datetime/dateinput) component.
We are working on making the Property Pane keyboard accessible, we have to manage its inputs events manually. Since there are a lot of customizations, here I am listing down all the changes.

1. Actions Bar:- The `Today` and `Cancel` buttons at the bottom of the Datepicker do not have
events handlers.
2. Time Picker:- Time picker has a Keydown event handler where we have to listen for en `Enter` key and
close the picker to select the time. At the component level, we already have a key-down handler.
3. Day grid:- We are listening for an event here `onDay` click.
4. The top "<" and ">" arrow controls:- On pressing `Space` and `Enter` they change the months but we cannot override them.
5. Month Select:- On pressing `Enter` nothing happens. On pressing `Space` it opens the select menu and then we can choose a month by pressing `Enter`.
6. Year Select:- On pressing `Enter` nothing happens. On pressing `Space` it opens the select menu and then we can choose a year by pressing `Enter`.

Basically what we want to discuss here is what to do for Actions Bar, what should be the behavior for pressing `Enter` for all of these 8 controls, and is it a good practice to manage so many controls manually.

### Steps To Reproduce

You can check the current deployment [here](https://appsmith-bkkruc3ph-get-appsmith.vercel.app/applications/622b30e65d6ad8344fae9514/pages/622b30e65d6ad8344fae9517/edit).

For the code changes please visit the [PR](https://github.com/appsmithorg/appsmith/pull/12190).

### Public Sample App

https://appsmith-bkkruc3ph-get-appsmith.vercel.app/applications/622b30e65d6ad8344fae9514/pages/622b30e65d6ad8344fae9517/edit

### Version

Cloud

**Comment by ankurrsinghal** [comment]
@vuiets @aswathkk @somangshu @rohitagarwal88
